hooks/honeybadger/honeybadger.go

```diff
@@ -0,0 +1,86 @@
+package logrus_honeybadger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+const apiURL = "https://api.honeybadger.io/v1/notices"
+
+// HoneybadgerHook to send exceptions to an exception-tracking service compatible
+// with the Honeybadger API.
+type HoneybadgerHook struct {
+	apiKey        string
+	hostname, env string
+	httpClient    *http.Client
+}
+
+func NewHoneyBadgerHook(apiKey, env string, timeout time.Duration) *HoneybadgerHook {
+	hostname, _ := os.Hostname()
+
+	return &HoneybadgerHook{
+		apiKey:     apiKey,
+		hostname:   hostname,
+		env:        env,
+		httpClient: &http.Client{Timeout: timeout},
+	}
+}
+
+func (hook *HoneybadgerHook) Post(message string) error {
+	payload, err := json.Marshal(NewPayload(hook.hostname, hook.env, message))
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", apiURL, bytes.NewReader(payload))
+	if err != nil {
+		return err
+	}
+	for k, v := range map[string]string{
+		"X-API-Key":    hook.apiKey,
+		"Content-Type": "application/json",
+		"Accept":       "application/json",
+	} {
+		req.Header.Set(k, v)
+	}
+
+	resp, err := hook.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
```

Comment:
What if it's not `nil` though? I don't know if `err` guarantees that the body doesn't need to be closed.